Upstream: Yes
From 3509cccd554fb0b495356afcf7a7a7e4dac0c624 Mon Sep 17 00:00:00 2001
From: Cyp <cyp@wz2100.net>
Date: Tue, 7 Mar 2017 09:44:25 +0100
Subject: [PATCH] Fix compilation error with some distros and/or library
 versions.

Reported on the forum thread ?t=12726:

editbox.cpp:127:9: error: ambiguous overload for 'operator==' (operand types are 'QChar' and 'char')
  if (ch == '\0')
---
 lib/widget/editbox.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/widget/editbox.cpp b/lib/widget/editbox.cpp
index 7856b1d..fc4b168 100644
--- a/lib/widget/editbox.cpp
+++ b/lib/widget/editbox.cpp
@@ -124,7 +124,7 @@ void W_EDITBOX::initialise()
 /* Insert a character into a text buffer */
 void W_EDITBOX::insertChar(QChar ch)
 {
-	if (ch == '\0')
+	if (ch == QChar('\0'))
 	{
 		return;
 	}
@@ -150,7 +150,7 @@ void W_EDITBOX::insertChar(QChar ch)
 /* Put a character into a text buffer overwriting any text under the cursor */
 void W_EDITBOX::overwriteChar(QChar ch)
 {
-	if (ch == '\0')
+	if (ch == QChar('\0'))
 	{
 		return;
 	}

